packages/svelte/src/compiler/phases/3-transform/css/index.js

```diff
@@ -112,49 +126,49 @@ const visitors = {
 			return;
 		}
 
-		const used = node.prelude.children.filter((s) => s.metadata.used);
-
-		if (used.length === 0) {
+		if (!node.prelude.children.some((s) => s.metadata.used)) {
 			state.code.prependRight(node.start, '/* (unused) ');
 			state.code.appendLeft(node.end, '*/');
 			escape_comment_close(node, state.code);
 
 			return;
 		}
 
-		if (used.length < node.prelude.children.length) {
-			let pruning = false;
-			let last = node.prelude.children[0].start;
+		next();
+	},
+	SelectorList(node, { state, next, path }) {
+		let pruning = false;
+		let last = node.children[0].start;
 
-			for (let i = 0; i < node.prelude.children.length; i += 1) {
-				const selector = node.prelude.children[i];
+		for (let i = 0; i < node.children.length; i += 1) {
+			const selector = node.children[i];
 
-				if (selector.metadata.used === pruning) {
-					if (pruning) {
-						let i = selector.start;
-						while (state.code.original[i] !== ',') i--;
+			if (selector.metadata.used === pruning) {
+				if (pruning) {
+					let i = selector.start;
+					while (state.code.original[i] !== ',') i--;
 
-						state.code.overwrite(i, i + 1, '*/');
+					state.code.overwrite(i, i + 1, '*/');
+				} else {
+					if (i === 0) {
+						state.code.prependRight(selector.start, '/* (unused) ');
 					} else {
-						if (i === 0) {
-							state.code.prependRight(selector.start, '/* (unused) ');
-						} else {
-							state.code.overwrite(last, selector.start, ' /* (unused) ');
-						}
+						state.code.overwrite(last, selector.start, ' /* (unused) ');
 					}
-
-					pruning = !pruning;
 				}
 
-				last = selector.end;
+				pruning = !pruning;
 			}
 
-			if (pruning) {
-				state.code.appendLeft(last, '*/');
-			}
+			last = selector.end;
 		}
 
-		next();
+		if (pruning) {
+			state.code.appendLeft(last, '*/');
+		}
+
+		const specificity = path.at(-1)?.type === 'Rule' ? { bumped: false } : state.specificity;
```

Comment:
do we want to avoid using at() in general or only in runtime code?  (https://github.com/sveltejs/svelte/issues/10438)

Reply:
we're using it throughout the compiler already